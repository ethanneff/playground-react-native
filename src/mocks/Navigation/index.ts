import { jest } from '@jest/globals';
export const mockNavigate = jest.fn();
export const mockGoBack = jest.fn();
