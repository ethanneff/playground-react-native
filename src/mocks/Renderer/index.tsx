import {
  BaseNavigationContainer,
  NavigationContext,
  NavigationRouteContext,
} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React, {ReactNode} from 'react';
import {Provider} from 'react-redux';
import {create, ReactTestRenderer} from 'react-test-renderer';
import {store} from '../../redux/core';

type MockNavigation = any;
type MockRenderer = {
  tree: ReactTestRenderer;
  navigation: MockNavigation;
};
type Props = {
  component: ReactNode;
  route?: {key: string; name: string};
  navigation?: MockNavigation;
};

export const mockRenderer = ({
  component,
  route = {key: '', name: ''},
  navigation = {
    goBack: jest.fn(),
    navigate: jest.fn(),
    push: jest.fn(),
    popToTop: jest.fn(),
    addListener: () => jest.fn(),
    isFocused: jest.fn().mockReturnValue(true),
    reset: jest.fn(),
    emit: jest.fn(),
  },
}: Props): MockRenderer => {
  const Stack = createStackNavigator();

  const Component = () => (
    <NavigationContext.Provider value={navigation}>
      <NavigationRouteContext.Provider value={route}>
        {component}
      </NavigationRouteContext.Provider>
    </NavigationContext.Provider>
  );

  const Providers = () => (
    <Provider store={store}>
      <BaseNavigationContainer>
        <Stack.Navigator>
          <Stack.Screen component={Component} name="MockedScreen" />
        </Stack.Navigator>
      </BaseNavigationContainer>
    </Provider>
  );

  const tree = create(<Providers />);

  return {tree, navigation};
};
