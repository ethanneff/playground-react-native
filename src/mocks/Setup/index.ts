import { NativeModules } from 'react-native';
import 'react-native-gesture-handler/jestSetup';

jest.mock('@invertase/react-native-apple-authentication', () => () => ({
  appleAuth: jest.fn(),
}));

const mockUserInfo = {
  idToken: 'mockIdToken',
  accessToken: null,
  accessTokenExpirationDate: null,
  serverAuthCode: 'mockServerAuthCode',
  scopes: [],
  user: {
    email: 'mockEmail',
    id: 'mockId',
    givenName: 'mockGivenName',
    familyName: 'mockFamilyName',
    photo: 'mockPhotoUtl',
    name: 'mockFullName',
  },
};

jest.mock('@react-native-community/google-signin', () => ({
  GoogleSignin: {
    configure: jest.fn(),
    hasPlayServices: jest.fn(() => Promise.resolve(true)),
    signIn: jest.fn(() => Promise.resolve(mockUserInfo)),
    signInSilently: jest.fn(() => Promise.resolve(mockUserInfo)),
    revokeAccess: jest.fn(() => Promise.resolve(true)),
    signOut: jest.fn(() => Promise.resolve(true)),
  },
}));

jest.mock('@react-native-firebase/analytics', () => () => ({
  logEvent: jest.fn(),
  setUserProperties: jest.fn(),
  setUserId: jest.fn(),
  setCurrentScreen: jest.fn(),
}));

jest.mock('react-native-sound', () => {
  return class Mock {
    constructor() {}
    setVolume = jest.fn();
    setNumberOfLoops = jest.fn();
    play = jest.fn();
    stop = jest.fn();
    static setCategory = jest.fn();
  };
});

jest.mock('react-native-device-info', () => ({
  getAPILevel: jest.fn(),
  getApplicationName: jest.fn(),
  getBatteryLevel: jest.fn(() => Promise.resolve(1)),
  getBrand: jest.fn(),
  getBuildNumber: jest.fn(),
  getBundleId: jest.fn(),
  getCarrier: jest.fn(),
  getDeviceCountry: jest.fn(),
  getDeviceId: jest.fn(),
  getDeviceLocale: jest.fn(),
  getDeviceName: jest.fn(),
  getFirstInstallTime: jest.fn(),
  getFontScale: jest.fn(),
  getFreeDiskStorage: jest.fn(),
  getInstallReferrer: jest.fn(),
  getInstanceID: jest.fn(),
  getLastUpdateTime: jest.fn(),
  getManufacturer: jest.fn(),
  getMaxMemory: jest.fn(),
  getModel: jest.fn(),
  getPhoneNumber: jest.fn(),
  getReadableVersion: jest.fn(),
  getSerialNumber: jest.fn(),
  getSystemName: jest.fn(),
  getSystemVersion: jest.fn(),
  getTimezone: jest.fn(() => 'America/Los_Angeles'),
  getTotalDiskCapacity: jest.fn(),
  getTotalMemory: jest.fn(),
  getUniqueID: jest.fn(),
  getUserAgent: jest.fn(),
  getVersion: jest.fn(),
  is24Hour: jest.fn(),
  isEmulator: jest.fn(),
  isPinOrFingerprintSet: jest.fn(),
  isTablet: jest.fn(() => false),
}));

NativeModules.RNCNetInfo = {
  addListener: jest.fn(),
  getCurrentState: jest.fn(() => Promise.resolve()),
  removeListeners: jest.fn(),
};

NativeModules.RNCAsyncStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeChecklistItem: jest.fn(),
  mergeItem: jest.fn(),
  clear: jest.fn(),
  getAllKeys: jest.fn(),
  flushGetRequests: jest.fn(),
  multiGet: jest.fn(),
  multiSet: jest.fn(),
  multiRemove: jest.fn(),
  multiMerge: jest.fn(),
};

jest.useFakeTimers();

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
