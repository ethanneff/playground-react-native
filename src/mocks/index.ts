export * from './Navigation';
export * from './Renderer';
