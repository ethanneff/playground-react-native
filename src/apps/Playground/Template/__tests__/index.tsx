import React from 'react';
import { Template } from '..';
import { mockRenderer } from '../../../../mocks/Renderer';

describe('template', () => {
  it('renders correctly', () => {
    expect.hasAssertions();
    const { tree } = mockRenderer({ component: <Template /> });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
