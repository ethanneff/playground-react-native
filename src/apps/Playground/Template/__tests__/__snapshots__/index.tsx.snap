// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "marginHorizontal": 16,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  MockedScreen
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "hsl(0,0%,100%)",
                            "flex": 1,
                          }
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "hsl(0,0%,100%)",
                                "borderBottomColor": "hsl(0,0%,100%)",
                                "borderBottomWidth": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                                "paddingHorizontal": 8,
                              }
                            }
                          >
                            <RNGestureHandlerButton
                              collapsable={false}
                              onGestureEvent={[Function]}
                              onGestureHandlerEvent={[Function]}
                              onGestureHandlerStateChange={[Function]}
                              onHandlerStateChange={[Function]}
                              rippleColor={0}
                              style={
                                Array [
                                  Object {
                                    "flex": 0,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                style={
                                  Object {
                                    "opacity": 1,
                                    "padding": 8,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      Object {
                                        "justifyContent": "center",
                                      },
                                      false,
                                    ]
                                  }
                                  testID="leftNav"
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "hsl(0,0%,32.9%)",
                                          "fontSize": 32,
                                        },
                                        Array [
                                          undefined,
                                          undefined,
                                        ],
                                        Object {
                                          "fontFamily": "Material Design Icons",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    󰅁
                                  </Text>
                                </View>
                              </View>
                            </RNGestureHandlerButton>
                            <Text
                              numberOfLines={1}
                              onPress={[Function]}
                              style={
                                Object {
                                  "color": "hsla(0,0%,0%, 1)",
                                  "flex": 1,
                                  "fontSize": 34,
                                  "fontWeight": "300",
                                  "letterSpacing": 0.25,
                                  "opacity": 1,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Template
                            </Text>
                            <RNGestureHandlerButton
                              collapsable={false}
                              onGestureEvent={[Function]}
                              onGestureHandlerEvent={[Function]}
                              onGestureHandlerStateChange={[Function]}
                              onHandlerStateChange={[Function]}
                              rippleColor={0}
                              style={
                                Array [
                                  Object {
                                    "flex": 0,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                style={
                                  Object {
                                    "opacity": 1,
                                    "padding": 8,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      Object {
                                        "justifyContent": "center",
                                      },
                                      Object {
                                        "alignSelf": "flex-end",
                                      },
                                    ]
                                  }
                                  testID="rightNav"
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "transparent",
                                          "fontSize": 32,
                                        },
                                        Array [
                                          undefined,
                                          undefined,
                                        ],
                                        Object {
                                          "fontFamily": "Material Design Icons",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    󰅖
                                  </Text>
                                </View>
                              </View>
                            </RNGestureHandlerButton>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "hsl(0,0%,100%)",
                                }
                              }
                            >
                              <View />
                            </View>
                          </View>
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
